ops/internal/manage/depsets.go

```diff
@@ -0,0 +1,243 @@
+		}
+		interopTime := *cfg.Config.Hardforks.InteropTime.U64Ptr()
+		if thisDepset[chainId].ActivationTime < interopTime {
+			return fmt.Errorf("%w: chainId %d, activation_time %d interop_time %d",
+				errInvalidActivationTime,
+				cfg.Config.ChainID,
+				thisDepset[chainId].ActivationTime,
+				interopTime)
+		}
+
+		// check deep equality of dependency sets
+		if i > 0 {
+			// 1. check if the maps have the same length
+			if len(firstDepset) != len(thisDepset) {
+				return fmt.Errorf("%w: chain %d has different number of dependencies than first chain (%d vs %d)",
+					errDepsetLengths,
+					cfg.Config.ChainID, len(thisDepset), len(firstDepset))
+			}
+
+			// 2. check for deep equality of dependency values
+			for chainID, dep := range firstDepset {
+				otherDep, exists := thisDepset[chainID]
+				if !exists {
+					return fmt.Errorf("%w: chain %d is missing dependency for chain ID %s that exists in first chain",
+						errInconsistentDepsets,
+						cfg.Config.ChainID, chainID)
+				}
+
+				if dep.ChainIndex != otherDep.ChainIndex {
+					return fmt.Errorf("%w: chain %d has different chain index for dependency %s (%d vs %d)",
+						errInconsistentDepsets,
+						cfg.Config.ChainID, chainID, otherDep.ChainIndex, dep.ChainIndex)
+				}
+				if dep.ActivationTime != otherDep.ActivationTime {
+					return fmt.Errorf("%w: chain %d has different activation time for dependency %s (%d vs %d)",
+						errInconsistentDepsets,
+						cfg.Config.ChainID, chainID, otherDep.ActivationTime, dep.ActivationTime)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// checkOnchain ensures that DisputeGameFactoryProxy and EthLockboxProxy addresses read from onchain
+// are the same for all chains in a depset
+func (dc *DepsetChecker) checkOnchain(cfgs []DiskChainConfig) error {
+	if len(cfgs) == 0 {
+		return fmt.Errorf("no chain configs provided to checkOnchain")
+	}
+
+	if len(cfgs) == 1 {
+		return nil
+	}
+
+	// Get and validate the first chain's addresses
+	firstAddrs, err := dc.getAndValidateAddresses(cfgs[0].Config.ChainID)
+	if err != nil {
+		return err
+	}
+
+	firstDisputeGameFactoryProxy := strings.ToLower((*firstAddrs.DisputeGameFactoryProxy).String())
+	// TODO: re-enable this once we can pull in the updated op-fetcher from monorepo
+	// firstEthLockboxProxy := strings.ToLower((*firstAddrs.EthLockboxProxy).String())
+
+	// Check all chains in the dependency set
+	for _, cfg := range cfgs {
+		addrs, err := dc.getAndValidateAddresses(cfg.Config.ChainID)
+		if err != nil {
+			return err
+		}
+
+		if strings.ToLower((*addrs.DisputeGameFactoryProxy).String()) != firstDisputeGameFactoryProxy {
+			return fmt.Errorf("DisputeGameFactoryProxy address mismatch for chain %d, expected %s, got %s",
+				cfg.Config.ChainID, firstDisputeGameFactoryProxy, addrs.DisputeGameFactoryProxy)
+		}
+		// TODO: re-enable this once we can pull in the updated op-fetcher from monorepo
```

Comment:
is it worth making a tracking issue and recording it here?

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1013/commits/c03558c4a594cd6850615cebd73e0f6a8960536e